File: test_data/java_programs/openjdk_testsuite/02322.java





class LambdaTest {

    interface I {
        int m(int x);
    }

    static int foo(I i) { return i.m(0); }

    static {
        foo((int x1) -> { return foo((int x2) -> { return x1 + x2; }); });
    }
}

File: test_data/c_programs/gcc_testsuite/02322.c






extern void exit (int);
extern void abort (void);

int
foo (void)
{
  exit (0);
  return 0;
}

void
bar (void)
{
}

int
main (void)
{
  ((long (*)(int))bar) (foo ());
  abort ();
}
